syntax = "proto3";

package student;

service StudentService {
  rpc CreateStudent (Student) returns (StudentResponse);
  rpc GetStudent (StudentRequest) returns (Student);
  rpc ListStudents (EmptyRequest) returns (StudentList);
}

message Student {
  string first_name = 1;
  string last_name = 2;
  int32 age = 3;
}

message StudentResponse {
  string message = 1;
}

message StudentRequest {
  int32 id = 1;
}

message StudentList {
  repeated Student students = 1;
}

message EmptyRequest {}
